bot/src/main/kotlin/org/epilink/bot/discord/LinkDiscord4JFacadeImpl.kt

```diff
@@ -58,18 +55,19 @@ internal class LinkDiscord4JFacadeImpl(
     /**
      * The actual Discord client
      */
-    private val client = DiscordClientBuilder.create(token).build().apply {
-        eventDispatcher.onEvent(MemberJoinEvent::class) { handle() }
-        eventDispatcher.onEvent(MessageCreateEvent::class) { handle() }
-    }
+    private var cclient: GatewayDiscordClient? = null
+
+    private val client
```

Comment:
Why not use 'lateinit val' which would achieve the same result with one line ?

Reply:
The exception in case of lateinit vars is unclear. I prefer controlling that sort of thing myself and clearly stating "hey, you forgot to start the thing". Also, you cannot check for initialization of lateinit vars in a clean way, whereas here it's as simple as a null check on `cclient`. Admittedly `cclient`'s name is not super clear but i'm lazy.